modelbaker/processing/ili2db_exporting.py

```diff
@@ -0,0 +1,439 @@
+            result = exporter.run(None)
+            if result == iliexporter.Exporter.SUCCESS:
+                feedback.pushInfo(self.tr("... export succeeded"))
+            else:
+                feedback.pushWarning(self.tr("... export failed"))
+            isvalid = bool(result == iliexporter.Exporter.SUCCESS)
+        except JavaNotFoundError as e:
+            raise QgsProcessingException(
+                self.tr("Java not found error:").format(e.error_string)
+            )
+
+        return {self.ISVALID: isvalid, self.XTFFILEPATH: configuration.xtffile}
+
+    def get_configuration_from_input(self, parameters, context, tool):
+
+        configuration = ExportConfiguration()
+        configuration.base_configuration = self.parent.current_baseconfig()
+        configuration.tool = tool
+
+        # get database settings form the parent
+        if not self.parent.get_db_configuration_from_input(
+            parameters, context, configuration
+        ):
+            return None
+
+        # get settings according to the db
+        configuration.with_exporttid = self._get_tid_handling(configuration)
+
+        # get settings from the input
+        filtermode = self.parent.parameterAsString(parameters, self.FILTERMODE, context)
+        filters = self.parent.parameterAsString(parameters, self.FILTER, context)
+        if filtermode == "Models" and filters:
+            configuration.ilimodels = filters
+        elif filtermode == "Datasets" and filters:
+            configuration.dataset = filters
+        elif filtermode == "Baskets" and filters:
+            configuration.baskets = filters
+        else:
+            configuration.ilimodels = ";".join(self._get_model_names(configuration))
+
+        exportmodels = self.parent.parameterAsString(
+            parameters, self.EXPORTMODEL, context
+        )
+        if exportmodels:
+            configuration.iliexportmodels = exportmodels
+
+        configuration.disable_validation = self.parent.parameterAsBool(
+            parameters, self.DISABLEVALIDATION, context
+        )
+
+        configuration.xtffile = self.parent.parameterAsFile(
+            parameters, self.XTFFILEPATH, context
+        )
+
+        return configuration
+
+    def _get_tid_handling(self, configuration):
+        db_connector = get_db_connector(configuration)
+        if db_connector:
+            return db_connector.get_tid_handling()
+        return False
+
+    def _get_model_names(self, configuration):
+        modelnames = []
+
+        db_connector = get_db_connector(configuration)
+        if (
+            db_connector
+            and db_connector.db_or_schema_exists()
+            and db_connector.metadata_exists()
+        ):
+            db_models = db_connector.get_models()
+            regex = re.compile(r"(?:\{[^\}]*\}|\s)")
+            for db_model in db_models:
+                for modelname in regex.split(db_model["modelname"]):
```

Comment:
Do you have an example of why we need this regex?
I thought we got a nice model name in `db_model["modelname"]`, like in [this test](https://github.com/opengisch/QgisModelBakerLibrary/blob/92c691cd9008183d114402f7fd1143cc3edc694a/tests/test_get_models.py#L8), although I imagine there are more complex scenarios.